MarkEditMac/Sources/Scripting/EditorDocument+Scripting.swift

```diff
@@ -0,0 +1,174 @@
+      let markdownOptions = AttributedString.MarkdownParsingOptions(
+        allowsExtendedAttributes: true,
+        interpretedSyntax: .full,
+        failurePolicy: .returnPartiallyParsedIfPossible,
+        appliesSourcePositionAttributes: true
+      )
+
+      // Process lines separately to ensure they stay distinct
+      let newlineString = NSAttributedString(string: "\n")
+      let attributedSource = self.source.components(separatedBy: .newlines)
+      .map { component in
+        do {
+          return try NSMutableAttributedString(markdown: component, options: markdownOptions)
+        } catch {
+          return NSMutableAttributedString(string: component)
+        }
+      }
+      .reduce(into: NSMutableAttributedString()) { result, element in
+        element.append(newlineString)
+        result.append(element)
+      }
+
+      return NSTextStorage(attributedString: attributedSource)
+  }
+
+  /// Executes a user-provided JS script in the document's webview.
+  @objc func handleEvaluateCommand(_ command: NSScriptCommand) -> Any? {
+    guard let inputString = command.evaluatedArguments?["script"] as? String else {
+      let scriptError = ScriptError.missingInput(message: "No JavaScript script provided")
+      command.scriptErrorString = scriptError.localizedDescription
+      command.scriptErrorNumber = scriptError.code
+      return nil
+    }
+
+    guard let targetEditor = self.targetEditor(for: command) else {
+      return nil
+    }
+
+    // Preserve newlines for better error reporting
+    let jsText = inputString.replacingOccurrences(of: "\\n", with: "\\\\n")
+
+    command.suspendExecution()
+    targetEditor.webView.evaluateJavaScript(jsText) { value, error in
+      if let error = error as NSError? {
+        let scriptError = ScriptError.jsEvalError(jsError: error)
+        command.scriptErrorString = scriptError.localizedDescription
+        command.scriptErrorNumber = scriptError.code
+        command.resumeExecution(withResult: error.localizedDescription)
+        return
+      }
+
+      // Pack JS value into an AS-compatible descriptor
+      let descriptor = NSAppleEventDescriptor(with: value)
+      command.resumeExecution(withResult: descriptor)
+    }
+    return nil
+  }
+}
+
+// MARK: - Private
+
+private extension EditorDocument {
+  func currentCommand() -> NSScriptCommand? {
+    guard let currentCommand = NSScriptCommand.current() else {
+      Logger.log(.error, "Couldn't find a command to handle incoming Apple Event.")
+      return nil
+    }
+    return currentCommand
+  }
+
+  func targetEditor(for command: NSScriptCommand?) -> EditorViewController? {
+    guard let currentCommand = command == nil ? currentCommand() : command else {
+      Logger.log(.error, "Couldn't find a command to handle incoming Apple Event.")
+      return nil
+    }
+
+    guard let targetEditor = self.windowControllers.first?.contentViewController as? EditorViewController else {
```

Comment:
nit: code convention wise, I prefer to hide `self.` if we can. I can do a cleanup later.